keanu-project/src/main/java/io/improbable/keanu/distributions/discrete/UniformInt.java

```diff
@@ -0,0 +1,43 @@
+package io.improbable.keanu.distributions.discrete;
+
+import io.improbable.keanu.distributions.DiscreteDistribution;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.tensor.intgr.IntegerTensor;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+
+public class UniformInt implements DiscreteDistribution {
+
+    private final IntegerTensor xMin;
+    private final IntegerTensor xMax;
+
+    public static DiscreteDistribution withParameters(IntegerTensor xMin, IntegerTensor xMax) {
+        return new UniformInt(xMin, xMax);
+    }
+
+    private UniformInt(IntegerTensor xMin, IntegerTensor xMax) {
+        this.xMin = xMin;
+        this.xMax = xMax;
+    }
+
+    @Override
+    public IntegerTensor sample(int[] shape, KeanuRandom random) {
+        DoubleTensor minDouble = xMin.toDouble();
+        DoubleTensor delta = xMax.toDouble().minus(minDouble);
+        DoubleTensor randoms = random.nextDouble(shape);
+
+        return delta.timesInPlace(randoms).plusInPlace(minDouble).toInteger();
+    }
+
+    @Override
+    public DoubleTensor logProb(IntegerTensor x) {
+        DoubleTensor maxBound = xMax.toDouble();
+        DoubleTensor minBound = xMin.toDouble();
+        DoubleTensor xDouble = x.toDouble();
+
+        DoubleTensor logOfWithinBounds = maxBound.minus(minBound).logInPlace().unaryMinusInPlace();
+        logOfWithinBounds = logOfWithinBounds.setWithMaskInPlace(xDouble.getGreaterThanMask(maxBound), Double.NEGATIVE_INFINITY);
+        logOfWithinBounds = logOfWithinBounds.setWithMaskInPlace(xDouble.getLessThanOrEqualToMask(minBound), Double.NEGATIVE_INFINITY);
+
```

Comment:
Might also be worth adding a test for the prob calculation as well?  It doesn't look like this is tested in UniformIntVertexTest at the moment?